plugins/omelasticsearch/omelasticsearch.c

```
@@ -1206,50 +1274,93 @@ curlResult(void *ptr, size_t size, size_t nmemb, void *userdata)
+}
 
+static void
+curlPostSetup(CURL *handle, HEADER *header, uchar* authBuf)
+{
+	curl_easy_setopt(handle, CURLOPT_HTTPHEADER, header);
 	curl_easy_setopt(handle, CURLOPT_WRITEFUNCTION, curlResult);
 	curl_easy_setopt(handle, CURLOPT_POST, 1);
 	curl_easy_setopt(handle, CURLOPT_NOSIGNAL, TRUE);
 
-	pWrkrData->curlHandle = handle;
-	pWrkrData->postHeader = header;
+	if(authBuf != NULL) {
+		curl_easy_setopt(handle, CURLOPT_USERPWD, authBuf);
+		curl_easy_setopt(handle, CURLOPT_PROXYAUTH, CURLAUTH_ANY);
+	}
+}
 
-	if(    pData->bulkmode
-	   || (pData->dynSrchIdx == 0 && pData->dynSrchType == 0 && pData->dynParent == 0)) {
-		/* in this case, we know no tpls are involved in the request-->NULL OK! */
-		setCurlURL(pWrkrData, pData, NULL);
+static rsRetVal
+curlSetup(wrkrInstanceData_t *pWrkrData, instanceData *pData)
+{
+	pWrkrData->curlHeader = curl_slist_append(NULL, "Content-Type: text/json; charset=utf-8");
+	pWrkrData->curlPostHandle = curl_easy_init();
+	if (pWrkrData->curlPostHandle == NULL) {
+		return RS_RET_OBJ_CREATION_FAILED;
 	}
+	curlPostSetup(pWrkrData->curlPostHandle, pWrkrData->curlHeader, pData->authBuf);
 
-	if(Debug) {
-		if(pData->dynSrchIdx == 0 && pData->dynSrchType == 0 && pData->dynParent == 0)
-			dbgprintf("omelasticsearch setup, using static REST URL\n");
-		else
-			dbgprintf("omelasticsearch setup, we have a dynamic REST URL\n");
+	pWrkrData->curlCheckConnHandle = curl_easy_init();
+	if (pWrkrData->curlCheckConnHandle == NULL) {
+		curl_easy_cleanup(pWrkrData->curlPostHandle);
+		pWrkrData->curlPostHandle = NULL;
+		return RS_RET_OBJ_CREATION_FAILED;
 	}
+	curlCheckConnSetup(pWrkrData->curlCheckConnHandle, pWrkrData->curlHeader, 
+		pData->healthCheckTimeout, pData->allowUnsignedCerts);
+
 	return RS_RET_OK;
 }
 
 static void
 setInstParamDefaults(instanceData *pData)
 {
-	pData->server = NULL;
-	pData->port = 9200;
+	pData->serverBaseUrls = NULL;
+	pData->defaultPort = 9200;
+	pData->healthCheckTimeout = 3500;
```

curl's default is infinite and elastic's default is 30 seconds, both of which seemed way too long if we're trying to process messages quickly and fail fast. 3.5 seconds seems like a reasonable default, that can be extended if users need.